### PR TITLE
Recipe App: Fix n+1 problems

### LINKED ISSUE #10: [ 1pt ] ⚠ Eliminate N+1 Problems 🔴
- [ ] install gem
- [ ] run all pages and fix (add includes) all queries as pointed

### PR DESCRIPTION
**Pull request by amare1990**
**While addressing this issue, I** 

- [x] Added `bullet` gem and install it in the development server.
- [x] Run the command `bundle exec rails g bullet:install`
- [x] Noticed that there is a query optimization required while we browse public recipes and seeing a recipe details.
- [x] Manually added `includes(:xxx)` in codes where query optimization occurs. 
- [x] Checked linters errors.

### CODE REVIEW

File: app/controllers/recipes_controller.rb [hunk 1]
```diff
@@ -7,10 +7,12 @@ def index
   end
 
   # GET /recipes/1 or /recipes/1.json
-  def show; end
+  def show
+    @recipefoods = RecipeFood.where(recipe_id: params[:id])
```
Comment (tiagomarin) [inline app/controllers/recipes_controller.rb]: When I clicked in the name of the ingredient in a recipe ( goes to recipe_food SHOW view ) I got a n1+ problem alert:

image

Also check recipe foods :
- new view
- index view 
- edit view

**Review by tiagomarin — CHANGES_REQUESTED** [review]
## STATUS: Changes Required :no_entry_sign: 

image

Hi @amare1990 

Thanks for all your contribution so far, all you did was correct :1st_place_medal:  , however you forgot to fix n1+ problems in some views. Check it out on my comments below :point_down: 

## To highlight :champagne: 
- Good commit history :heavy_check_mark: 
- Good PR description :heavy_check_mark: 
- You didn't have mistakes, only forgot some things :heavy_check_mark:  

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

File: app/controllers/recipes_controller.rb (hunk 1 above)
Comment (amare1990) [reply app/controllers/recipes_controller.rb]: Thanks @tiagomarin for reviewing working on my issues and the fantastic suggestions. I forgot checking `recipe_foods` component. I will work on it now.

Comment (amare1990) [reply app/controllers/recipes_controller.rb]: Hi @tiagomarin . I have made changes based on the suggestions you gave me. Kindly review my code again please.
Thanks

**Review by tiagomarin — APPROVED** [review]
# STATUS: APPROVE :heavy_check_mark: :heavy_check_mark:  :champagne: 

Hi @amare1990 
This step is complete!  Great work! 

Good to go! :shipit:  Congratulations! 🎉 :clap: :clap: :100: 

nodding-yes

## To highlight :champagne: 
- Good commit history :heavy_check_mark: 
- Good PR description :heavy_check_mark: 
- All n1+ problems are gone  :heavy_check_mark: 

#### Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**Pull request merged**